templatestore/frontend/src/components/TinyUrlComponent.js

```diff
@@ -0,0 +1,189 @@
+import React, { useState, useEffect } from 'react';
+import styles from '../style/templateScreen.less';
+import axios from 'axios';
+import { backendSettings } from './../utils.js';
+import TinyUrlDropdownComponent from './TinyUrlDropdownComponent';
+
+export default function TinyUrlComponent(props) {
+    if (props.contextData === '') return <></>;
+    var tinyUrlObjLocal = props.tinyUrlObj;
+    const [items, setItems] = useState([{ urlKey: "", expiry: "" }]);
```

Comment:
Major: We must avoid creating internal state from the props. it's a bad practice.
Either call the API in this component or just dont create state here, update the prop using updater functions which should be sent from parent component.